pkg/koordlet/util/resourceexecutor/updater.go

```diff
@@ -297,6 +314,31 @@ func CommonDefaultUpdateFunc(resource ResourceUpdater) error {
 	return sysutil.CommonFileWriteIfDifferent(c.Path(), c.value)
 }
 
+func CgroupUpdateWithUnlimitedFunc(resource ResourceUpdater) error {
+	c := resource.(*CgroupResourceUpdater)
+	// NOTE: convert "-1" to "max", since some cgroups-v2 files only accept "max" to unlimit resource instead of "-1".
+	//       DO NOT use it on the cgroups which has a valid value of "-1".
+	if c.value == "-1" && sysutil.GetCurrentCgroupVersion() == sysutil.CgroupVersionV2 {
+		c.value = "max"
+	}
+	_ = audit.V(5).Reason(ReasonUpdateCgroups).Message("update %v to %v", resource.Path(), resource.Value()).Do()
+	return sysutil.CgroupFileWriteIfDifferent(c.parentDir, c.file, c.value)
```

Comment:
//TODO mv function like CgroupFileWrite into resourceexecutor and set as private, so new developers won't call it directly

Reply:
the TODOs will be added in #879.